src/methods/SANA.cpp

```diff
@@ -1283,7 +1293,27 @@ void SANA::performChange(int type) {
         currentScore     = newCurrentScore;
 		exposedEdgesNumer= newExposedEdgesNumer;
         squaredAligEdges = newSquaredAligEdges;
+        MS3Numer         = newMS3Numer;
+        if (needMS3)
+        {
+            // score is calculated with denom 1 step behind if we do this here I guess
+            unsigned oldTarDeg, newTarDeg;
+            oldTarDeg = MultiS3::shadowDegrees[oldTarget];
+            newTarDeg = MultiS3::shadowDegrees[newTarget];
+
+            MultiS3::shadowDegrees[oldTarget] -= G1AdjLists[source].size();
+            if (oldTarDeg > 0 && !MultiS3::shadowDegrees[oldTarget])
```

Comment:
Uh.... I don't get this.  Didn't you just assign oldTarDegree to be exactly MultiS3::shadowDegrees[oldTarget]?